### ISSUE TITLE
Question about fine-tuning DeiT

### ISSUE THREAD

**Issue by NielsRogge** [facebookresearch/deit#82]
Hi @TouvronHugo,

I have a few questions regarding fine-tuning the distilled models. I would like to make sure I understand it correctly. So, there are 2 ways of fine-tuning a pre-trained distilled model, right?

1)  in the "classic" manner, by only placing a linear layer on top of the final hidden state of the [CLS] token, and not taking into account the final hidden state of the distillation token. Here, we perform regular cross-entropy between the prediction of the classifier and the ground-truth labels. This is similar to fine-tuning ViT.
2) by placing a linear layer on top of the final hidden state of the [CLS] token, **and** a linear layer on top of the final hidden state of the distillation token. Here, we perform regular cross-entropy between the prediction of the CLS classifier and the ground-truth class labels, and cross-entropy between the prediction of the distillation classifier and the label predicted by the teacher (assuming hard distillation). During inference, one takes the average prediction between both classifiers. 

(Based on https://github.com/facebookresearch/deit/issues/45#issuecomment-769655584).

Is this correct?

Thanks!

**Comment by TouvronHugo** [comment]
Hi @NielsRogge ,
Yes, that's absolutely right
Best,
Hugo

**Comment by NielsRogge** [comment]
Great! Thanks.
